CMSgov/design-system
# [WNMGDS-2338] Improve VoiceOver support for Dropdown

## Summary

WNMGDS-2338

- Manually creates a message in an aria-live region (through Downshift's API) that reads out options being highlighted to get around a [known VoiceOver issue](https://a11ysupport.io/tests/apg__aria-1-2-combobox-with-list-autocomplete-example#assertion-aria-aria-activedescendant_attribute-convey_value-aria-combobox_role)
- Removes `ariaLabel` as an explicit optional prop on `Dropdown` because it complicates things, and users can still add an `aria-label` attribute if they want because of the automatic pass-through
- Removes the `aria-labelledby` from the `listbox` because it was overly verbose and redundant
- Works around an issue where [buttons cannot expose both label and values at the same time](https://www.24a11y.com/2019/select-your-poison/)

## How to test

For the VoiceOver fix,

1. Open the default story in Safari
2. Turn on VoiceOver
3. Focus on the select
4. Use the down and up arrows to move through items. When it's done with its initial readout of the label, it should start to announce the item being highlighted

### To do
- [ ] Run this updated version by an accessibility expert

[Demo url](https://cmsgov.github.io/design-system/branch/pwolfert/dropdown-a11y-fixes/storybook/?path=/story/components-dropdown--default&globals=theme:healthcare)

## Checklist




### If this is a change to code:
